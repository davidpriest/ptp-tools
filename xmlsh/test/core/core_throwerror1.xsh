# throw on error basicsd


set -e

# simple one command 
try 
{ 
	true ;
	echo Success - executed true
	false ; 
	echo SNH Executed false
} catch X 
{ 
	echo Success caught error
} 

# dont throw when protected by conditions
try 
{ 
	! true
	! false 
	true && false || true
	if false ; then echo SNH ; fi
	X=<[ 0 ]>
	while [ $X -lt 1 ] ; do
		X=<[ $X + 1 ]>
	done
	
	echo Success 

} catch X 
{ 
	echo Failed should not have any errors 
} 

# brace commands with failures
try 
{ 
	{ true ; true ; false ;  } 
	echo SNH Executed false
} catch X 
{ 
	echo Success caught error
} 


# subshell commands with failures
try 
{ 
	( true ; true ; false ; echo Fail  )
	echo SNH Executed false
} catch X 
{ 
	echo Success caught error
} 

try 
{ 
	set +e 
	false 
	echo Success ignored error
} catch X 
{ 
	echo FAIL should not throw error
} 

