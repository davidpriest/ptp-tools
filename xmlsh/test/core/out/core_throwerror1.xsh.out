Success - executed true
Success caught error
Success
Success caught error
Success caught error
Success ignored error
